packages/rum-core/src/common/apm-server.js

```diff
@@ -193,70 +190,38 @@ class ApmServer {
       })
   }
 
-  initErrorQueue() {
-    if (this.errorQueue) {
-      this.errorQueue.flush()
-    }
-    this.errorQueue = this._createQueue(errors => {
-      var p = this.sendErrors(errors)
-      if (p) {
-        p.catch(reason => {
-          this._loggingService.warn(
-            'Failed sending errors!',
-            this._constructError(reason)
-          )
-        })
-      }
-    })
-
-    var limit = this._configService.get('errorThrottleLimit')
-    var interval = this._configService.get('errorThrottleInterval')
-
-    this.throttleAddError = throttle(
-      this.errorQueue.add.bind(this.errorQueue),
-      () => this._loggingService.warn('Dropped error due to throttling!'),
-      { limit, interval }
-    )
-  }
-
-  initTransactionQueue() {
-    if (this.transactionQueue) {
-      this.transactionQueue.flush()
+  createMetaData() {
+    const cfg = this._configService
+    const metadata = {
+      service: {
+        name: cfg.get('serviceName'),
+        version: cfg.get('serviceVersion'),
+        agent: {
+          name: 'js-base',
+          version: cfg.version
+        },
+        language: {
+          name: 'javascript'
+        },
+        environment: cfg.get('environment')
+      },
+      labels: cfg.get('context.tags')
     }
-    this.transactionQueue = this._createQueue(transactions => {
-      var p = this.sendTransactions(transactions)
-      if (p) {
-        p.catch(reason => {
-          this._loggingService.warn(
-            'Failed sending transactions!',
-            this._constructError(reason)
-          )
-        })
-      }
-    })
-
-    var limit = this._configService.get('transactionThrottleLimit')
-    var interval = this._configService.get('transactionThrottleInterval')
-
-    this.throttleAddTransaction = throttle(
-      this.transactionQueue.add.bind(this.transactionQueue),
-      () => this._loggingService.warn('Dropped transaction due to throttling!'),
-      { limit, interval }
-    )
+    return truncateModel(METADATA_MODEL, metadata)
   }
 
   addError(error) {
-    if (!this.errorQueue) {
-      this.initErrorQueue()
-    }
-    this.throttleAddError(error)
+    this.throttleEvents({
+      data: error,
+      type: ERRORS
+    })
   }
 
   addTransaction(transaction) {
-    if (!this.transactionQueue) {
-      this.initTransactionQueue()
-    }
-    this.throttleAddTransaction(transaction)
+    this.throttleEvents({
+      data: transaction,
+      type: TRANSACTIONS
```

Comment:
As discussed we can just use the type as the key: `{[type]:data}`

Reply:
I will also see if we can eliminate type and use the key for figuring out the event type.